kedro/framework/project/__init__.py

```diff
@@ -181,11 +179,6 @@ def configure_project(package_name: str):
     settings_module = f"{package_name}.settings"
     settings.configure(settings_module)
 
-    # set up all hooks so we can discover all pipelines
-    hook_manager = get_hook_manager()
-    _register_hooks(hook_manager, settings.HOOKS)
-    _register_hooks_setuptools(hook_manager, settings.DISABLE_HOOKS_FOR_PLUGINS)
```

Comment:
Moved this logic to happen when a new session is instantiated instead.

Reply:
My theory is that this is why the code fails when running with `ParallelRunner`. When we use `ParallelRunner`, we make sure each subprocess looks the same, even in `spawn` mode. That's why we call `_bootstrap_subprocess`, which configures a) the logging to be same and b) the project, via `configure_project`. `configure_project` also ensured that the hook manager has all the right hooks set up. Now that we don't do this here anymore, I believe the code should be replicated in `_bootstrap_process`. 
Not entirely sure why the `pluggy` hook manager doesn't get pickled/unpickled to the same object.

Reply:
Yes this is exactly what's happening! The reason why is that even though on first try it looked like the `PluginManager` could be pickled with the latest `pluggy` version, that wasn't actually the case.